dripline/implementations/entity_endpoints.py

```diff
@@ -0,0 +1,175 @@
+    '''
+
+    def __init__(self,
+                 get_str=None,
+                 get_reply_float=False,
+                 set_str=None,
+                 set_value_lowercase=True,
+                 set_value_map=None,
+                 extract_raw_regex=None,
+                 **kwargs):
+        '''
+        get_str (str): sent verbatim in the event of on_get; if None, getting of endpoint is disabled
+        get_reply_float (bool): apply special formatting to get return
+        set_str (str): sent as set_str.format(value) in the event of on_set; if None, setting of endpoint is disabled
+        set_value_lowercase (bool): default option to map all string set value to .lower()
+            **WARNING: never set to False if using a set_value_map dict
+        set_value_map (str||dict): inverse of calibration to map raw set value to value sent; either a dictionary or an asteval-interpretable string
+        extract_raw_regex (str): regular expression search pattern applied to get return. Must be constructed with an extraction group keyed with the name "value_raw" (ie r'(?P<value_raw>)' ) 
+        '''
+        Entity.__init__(self, **kwargs)
+        self._get_reply_float = get_reply_float
+        self._get_str = get_str
+        self._set_str = set_str
+        self._set_value_map = set_value_map
+        self._extract_raw_regex = extract_raw_regex
+        self.evaluator = asteval.Interpreter()
+        if set_value_map is not None and not isinstance(set_value_map, (dict,str)):
+            # exceptions.DriplineValueError
+            raise ValueError("Invalid set_value_map config for {}; type is {} not dict".format(self.name, type(set_value_map)))
+        self._set_value_lowercase = set_value_lowercase
+        if isinstance(set_value_map, dict) and not set_value_lowercase:
+            # exceptions.DriplineValueError
+            raise ValueError("Invalid config option for {} with set_value_map and set_value_lowercase=False".format(self.name))
+
+    @calibrate()
+    def on_get(self):
+        if self._get_str is None:
+            # exceptions.DriplineMethodNotSupportedError
+            raise Exception('<{}> has no get string available'.format(self.name))
+        result = self.service.send_to_device([self._get_str])
+        # logger.debug
+        print('result is: {}'.format(result))
+        if self._extract_raw_regex is not None:
+            first_result = result
+            matches = re.search(self._extract_raw_regex, first_result)
+            if matches is None:
+                # logger.error
+                print('matching returned none')
+                # exceptions.DriplineValueError
+                raise ValueError('returned result [{}] has no match to input regex [{}]'.format(first_result, self._extract_raw_regex))
+            # logger.debug
+            print("matches are: {}".format(matches.groupdict()))
+            result = matches.groupdict()['value_raw']
+        if self._get_reply_float:
```

Comment:
wasn't there a plan to phase out the get_reply_float because the regex is more generic?

Reply:
yeah, I completely forgot, will drop this.